code/processes/cmedecoder.q

```diff
@@ -8,16 +8,17 @@
 / process one message from log
 msg:{
   / generate dictionary from message, with correct tags & properly typed values
-  msg:(!/) flip {
+  msg:(!/) {
     / get field name & type from tag number
-    d:exec from .fix.fields where number=x[0];
+    d:.fix.fields each x[;0];
     / check if field has enumerations
-    enum:exec from .fix.enums where name=d[`name];
-    val:$[null enum[`name];x[1];enum[`values]@enum[`enums]?x[1]];
+    enum:.fix.enums each d[`name];
+    val:{$[null x[`name1];y[1];x[`values]@x[`enums]?y[1]]}'[enum;x];
     / fix field value type
-    val:.fix.typefuncs[d[`fixtype]] val;
+    val:{.fix.typefuncs[x] y}'[d[`fixtype];val];
```

Comment:
I think you can use something like this here:

``.fix.typefuncs[d`fixtype]@'val``

Reply:
```
q.cme))(.fix.typefuncs[d`fixtype]@'val)~{.fix.typefuncs[x] y}'[d[`fixtype];val]
1b
```

Reply:
```
q.cme))\ts:10000 .fix.typefuncs[d`fixtype]@'val
184 1312
q.cme))\ts:10000 {.fix.typefuncs[x] y}'[d[`fixtype];val]
257 1760
```